docs/bare_metal/c_small_libs.md

```diff
@@ -62,6 +64,29 @@ After you have completed the steps above, add `small` to the `supported_c_libs`
 }
 ```
 
+### Non-returning main()
+
+Arm microlib doesn't support exiting from `main()`; attempting to exit from `main()` causes a bare metal application to crash. Here we show two ways to prevent this.
+
+#### Sleeping in a loop
+
+One recommended technique is to sleep in a loop at the end of `main()`:
+```
+while (true) {
+    sleep();
+}
+```
+
+This is energy-efficient compared to an empty `while (true) {}` loop, which keeps the processor running. A loop is still needed, because `sleep()` returns after the system is woken up by an interrupt.
+
+#### Dispatching an EventQueue
+
+If your application is based on an `EventQueue`, dispatching it at the end of `main()` works as well:
+
+[![View code](https://www.mbed.com/embed/?url=https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/APIs_RTOS/EventQueue_ex_2/)](https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/APIs_RTOS/EventQueue_ex_2/main.cpp)
+
+The call to `queue.dispatch_forever()` never returns, as long as we don't break the dispatch anywhere. The `EventQueue` class puts the system to sleep to save energy between events.
+
 ### Note on uARM toolchain
 
-The uARM toolchain is the ARMC6 toolchain with the Arm microlib, the C micro-library. This toolchain will be deprecated after 5.15.
+The uARM toolchain is the ARMC5 toolchain with the Arm microlib, the C micro-library. This toolchain will be deprecated after 5.15.
```

Comment:
Fixing a pretty major typo...

Reply:
This was correct. Originally uARM toolchain was using ARMC5 but since ARMC5 was deprecated it was using ARMC6 https://github.com/ARMmbed/mbed-os/blob/master/tools/build_api.py/#L199

Reply:
Ah didn't know there was a quirk to point uARM to ARMC6. I'll change this sentence back.